### PR TITLE
docs: fix grammatical errors

### PR DESCRIPTION
**Pull request by cratiu222**
This PR corrects several grammatical errors in the documentation and comments, improving readability and clarity. These changes enhance the overall quality of the project’s written content.

Hope it helps.

### CODE REVIEW

File: tachyon/base/containers/circular_deque.h
Comment (ashjeong) [inline tachyon/base/containers/circular_deque.h:1094]: Since "adds complexity" is the common term used in coding, I think "adds **a significant amount of** extra complexity" would be better.

**Review by ashjeong — COMMENTED** [review]
Thanks for the contribution! I gave a few comments, but besides that looks good~

**Comment by cratiu222** [comment]
solved the comments

**Comment by chokobole** [comment]
@cratiu222 Sorry for the late reply. Please follow [the commit convention](https://github.com/kroma-network/.github/blob/main/CONTRIBUTING.md#commit-messages)

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM but let's fix the commit messages before merging

**Comment by cratiu222** [comment]
> LGTM but let's fix the commit messages before merging

what needs to be fixed?

**Comment by dongchangYoo** [comment]
> > LGTM but let's fix the commit messages before merging
> 
> what needs to be fixed?

We require relatively strict conventions for the users of the code.
For example, in the case of a `fix process_handle.h` commit, `chore(docs): fix typo` or 'chore(docs): fix typo in process_handle.h' would be appropriate.

Additionally, the last two commits should be squashed into one since they modify the same file with the same content.

**Comment by batzor** [comment]
@chokobole @dongchangYoo I think we can squash merge this? If not let's close this PR as stale.

**Comment by chokobole** [comment]
Close this PR, since it hasn't been updated so long.

**Pull request closed without merging**
